### PR TITLE
Pagination for business table

### PR DESCRIPTION
**Pull request by tristancusiCGI**
Overall this change fixes a performance issue by implementing pagination
for businesses listed in the business table. Previously all rows were
returned and filtering was done in the frontend but this caused
performance issues. So this changes the frontend by removing its ability
to filter on the entire dataset and instead requests the backend to only
send the rows needed at that time. This changes the backend by having it
perform pagination and filtering.

BACKEND
* added new search query parameter for backend called "reports" which
  signals which of these three options to filter on:
      -`all`: all businesses (aka no filter)
      -`complete`: all businesses where all locations have completed their
      reports (also businesses without locations?)
      -`outstanding`: all businesses where some locations have outstanding
      reports

* fixed SQL for ordering rows by specified column

* added SQL filters for filtering businesses with outstanding or
  completed reports

* added an enum for the business report filter options

FRONTEND
* added reports search query parameter functionality. If user selected
  filter radio button then the filter parameter is passed to the backend

* fixed pagination, implementation was there but wasnt used and needed
  tweaking

* removed `completed`, and `notCompleted` arrays and their creation from `businessList` as those
  are no longer calculated in the frontend. Instead the returned list of
  businesses will match what the user selected with the radio buttons and
  other parameters

* fixed "Clear All Filters" button to also clear reports filter and reset
  pagination

* changed radio buttons to now send filter selection to backend instead
  of changing `businessList` in frontend

* added a reset of some of the search parameter values on refresh to match
  on-prem functionality

* changed `BusinessFilter` enum values to better match backend

### CODE REVIEW

File: packages/bcer-data-portal/app/src/views/Dashboard/Overview.tsx [hunk 1]
```diff
@@ -25,9 +25,21 @@ function Dashboard() {
...
     onChangeSearch,
-    clearAllFilters
+    clearAllFilters,
+    totalRowCount,
+    searchOptions
   } = useBusiness();
 
+  // resets values on refresh
+  useEffect(() => {
+    console.log('resetting search options')
```
Comment (ahoplock) [inline packages/bcer-data-portal/app/src/views/Dashboard/Overview.tsx:35]: This should probably be commented out.

**Review by ahoplock — CHANGES_REQUESTED** [review]
Please see my comments.

File: packages/bcer-api/app/src/business/business.service.ts [hunk 2]
```diff
@@ -147,22 +148,52 @@ export class BusinessService {
...
+    const {search, category, reports, page, pageSize} = query || {};
+
     const qb = this.businessRepository.createQueryBuilder('business');
     qb.leftJoinAndSelect('business.locations', 'location')
       .leftJoinAndSelect('location.noi', 'noi')
       .loadRelationCountAndMap('location.salesCount', 'location.sales')
       .loadRelationCountAndMap('location.productsCount', 'location.products')
       .loadRelationCountAndMap('location.manufacturesCount', 'location.manufactures')
       .where("coalesce(business.legalName, '') != ''")
+
```
[the remark this answers is not in the document]
Comment (tristancusiCGI) [reply packages/bcer-api/app/src/business/business.service.ts:160]: gave it a go but got in incorrect query results. Going to leave it as is for now.

**Review by ahoplock — APPROVED** [review]

**Review by ahoplock — APPROVED** [review]
Reviewed and approved; will deploy to Dev/Test for further testing.

**Pull request merged**